src/GT.Core/Services/Impl/GTExperienceLevelService.cs

```diff
@@ -108,5 +124,68 @@ public async Task<bool> ExistsByNameAsync(string name)
 				return null;
 			}
 		}
+
+		public async Task<ExperienceLevelDTO?> GetByIdAsync(string id)
+		{
+			try
+			{
+				// Get entity
+				var entity = await _experienceLevelRepository
+					.GetAll()
+					.FirstOrDefaultAsync(e => e.Id == id);
+
+				if (entity == null)
+				{
+					_logger.LogInformation($"Entity with id {id} not found.");
+					return null;
+				}
+
+				// Map entity to DTO
+				var experienceLevelDTO = new ExperienceLevelDTO()
+				{
+					Id = entity.Id,
+					Name = entity.Name,
+				};
+
+				return experienceLevelDTO;
+			}
+			catch (Exception e)
+			{
+				_logger.LogError(e.Message);
+				return null;
+			}
+		}
+
+		public async Task UpdateAsync(ExperienceLevelDTO experienceLevelDTO, string name)
```

Comment:
Any specific reason we update by name and not id?

Reply:
This was a legacy from old code when the service was built. PO have created a card to go through all services regarding this in the backlog